app.rb

```diff
@@ -7,13 +7,21 @@ class App
   attr_accessor :books, :people, :rentals
 
   def initialize
-    @books = []
-    @people = []
-    @rentals = []
+    created_books_file = create_new_json('books')
+    @books = created_books_file ? JSON.parse(created_books_file.read, create_additions: true) : []
+    created_people_file = create_new_json('people')
+    @people = created_people_file ? JSON.parse(created_people_file.read, create_additions: true) : []
+    created_rent_file = create_new_json('rentals')
+    @rentals = created_rent_file ? JSON.parse(created_rent_file.read, create_additions: true) : []
+  end
+
+  def create_new_json(file_name)
+    File.write("#{file_name}.json", []) unless File.exist?("#{file_name}.json")
+    File.open("#{file_name}.json", 'r')
```

Comment:
as required you are to save the data on the exit of the app  but currently, your data saves while the app is still open

![image](https://github.com/KUBAHO3/oop-school-library/assets/60612329/8491e2ec-132e-45c8-b1b8-b8eb1c7a4934)

Reply:
Dear @alexander16108  I think the idea is about maintaining data after quitting the application so I did it to make our app a real-time application so that even if the power is cut before quitting the data remain untouched!